cmd/osbuild-worker/jobimpl-osbuild.go

```diff
@@ -119,6 +118,9 @@ func (impl *OSBuildJobImpl) Run(job worker.Job) error {
 	if err != nil {
 		return err
 	}
+	if osbuildOutput.Success == false {
+		panic("Osbuild failed, yet did not return an error. This is an implementation bug.")
```

Comment:
I'm not sure about this. From `RunOSBuild()`:


```golang
err = cmd.Wait()

// try to decode the output even though the job could have failed
var result osbuild.Result
decodeErr := json.Unmarshal(stdoutBuffer.Bytes(), &result)
if decodeErr != nil {
	return nil, fmt.Errorf("error decoding osbuild output: %v\nthe raw output:\n%s", decodeErr, stdoutBuffer.String())
}

if err != nil {
	// ignore ExitError if output could be decoded correctly
	if _, isExitError := err.(*exec.ExitError); !isExitError {
		return nil, fmt.Errorf("running osbuild failed: %v", err)
	}
}

return &result, nil
```

If osbuild returns non-zero but its output can be parsed (i.e. it failed to build the given manifest), then we should report this failure back to composer.

Reply:
Yes, I got confused with the code. After reading the commit message again, I get it.